build_tools/aws-sdk-code-generator/lib/aws-sdk-code-generator/shared_example.rb

```diff
@@ -26,7 +26,12 @@ def to_str_output
 
     def entry(json, ref, indent, path)
       case ref['type']
-      when 'structure' then structure(json, ref, indent, path)
+      when 'structure'
+        if ref['members'].keys == ["S", "N", "B", "SS", "NS", "BS", "M", "L", "NULL", "BOOL"] && @api['metadata']['endpointPrefix'] == 'dynamodb'
```

Comment:
if we are using metadata for identifying dynamo, how about a value that's unlikely to change and unique? I'd prefer service name related values or service id https://github.com/aws/aws-sdk-ruby/blob/master/apis/dynamodb/2012-08-10/api-2.json#L3

Although very rarely but it did happen that endpoint prefix changed for service models

Reply:
I feel like this particular one is pretty stable, but I agree this is the line that feels the most hacky in the entire change. I'm going to apply a small refactor...